app/src/main/java/com/dessalines/thumbkey/db/AppDb.kt

```diff
@@ -63,14 +66,25 @@ const val DEFAULT_KEY_MODIFICATIONS = ""
 data class AppSettings(
     @PrimaryKey(autoGenerate = true) val id: Int,
     @ColumnInfo(
-        name = "key_size",
-        defaultValue = DEFAULT_KEY_SIZE.toString(),
+        name = "auto_size_keys",
+        defaultValue = DEFAULT_AUTO_SIZE_KEYS.toString(),
     )
-    val keySize: Int,
+    val autoSizeKeys: Int,
+    @ColumnInfo(
+        name = "non_square_keys",
+        defaultValue = DEFAULT_NON_SQUARE_KEYS.toString(),
+    )
+    val nonSquareKeys: Int,
+    @ColumnInfo(
+        name = "key_height",
+        defaultValue = DEFAULT_KEY_HEIGHT.toString(),
+    )
+    val keyHeight: Int,
     @ColumnInfo(
         name = "key_width",
+        defaultValue = DEFAULT_KEY_WIDTH.toString(),
     )
-    val keyWidth: Int?,
+    val keyWidth: Int,
```

Comment:
I have a feeling going from optional to required, and adding a default value, is going to break the migration. Probably safer to just add a new column at the end and maybe rename this one to `key_width_old` (SQL for android doesn't let you drop columns unless you want to recreate the whole table, which is annoying).